packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -682,160 +948,607 @@ class _SortData {
+///  * [FocusTraversalOrder], a widget that assigns an order to a widget subtree
+///    for the [OrderedFocusTraversalPolicy] to use.
+class NumericFocusOrder extends FocusOrder {
+  /// Const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const NumericFocusOrder(this.order) : assert(order != null);
+
+  /// The numerical order to assign to the widget subtree using
+  /// [FocusTraversalOrder].
+  ///
+  /// Determines the placement of this widget in a sequence of widgets that defines
+  /// the order in which this node is traversed by the focus policy.
+  ///
+  /// Lower values will be traversed first.
+  final double order;
+
+  @override
+  int doCompare(NumericFocusOrder other) => order.compareTo(other.order);
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('order', order));
+  }
+}
+
+/// Can be given to a [FocusTraversalOrder] widget to use a String to assign a
+/// lexical order to a widget subtree that is using a
+/// [OrderedTraversalPolicy] to define the order in which widgets should be
+/// traversed with the keyboard.
+///
+/// This sorts strings using Dart's default string comparison, which is not
+/// locale specific.
+///
+/// {@macro flutter.widgets.focusorder.comparable}
+///
+/// See also:
+///
+///  * [FocusTraversalOrder], a widget that assigns an order to a widget subtree
+///    for the [OrderedFocusTraversalPolicy] to use.
+class LexicalFocusOrder extends FocusOrder {
+  /// Const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const LexicalFocusOrder(this.order) : assert(order != null);
+
+  /// The String that defines the lexical order to assign to the widget subtree
+  /// using [FocusTraversalOrder].
+  ///
+  /// Determines the placement of this widget in a sequence of widgets that defines
+  /// the order in which this node is traversed by the focus policy.
+  ///
+  /// Lower lexical values will be traversed first (e.g. 'a' comes before 'z').
+  final String order;
+
+  @override
+  int doCompare(LexicalFocusOrder other) => order.compareTo(other.order);
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(StringProperty('order', order));
+  }
+}
+
+// Used to help sort the focus nodes in an OrderedFocusTraversalPolicy.
+class _OrderedFocusInfo {
+  const _OrderedFocusInfo({@required this.node, @required this.order})
+      : assert(node != null),
+        assert(order != null);
+
+  final FocusNode node;
+  final FocusOrder order;
+}
+
+/// A [FocusTraversalPolicy] that orders nodes by an explicit order that resides
+/// in the nearest [FocusTraversalOrder] widget ancestor.
+///
+/// {@macro flutter.widgets.focusorder.comparable}
```

Comment:
Turn this into a fancy dartpad sample? :)

Reply:
Oooh, good idea.  I wrote this before I had converted them.